### ISSUE TITLE
[FEATURE REQUEST] PUSHCUT Webhook Notification

### ISSUE THREAD

**Issue by monza258** [potrudeau/homebridge-messenger#5]
Hey, 
I'm curious to see what you've planned. Nice Plugin 

I would prefer a pushcut intigration. Is that planned? Pushcut is at the beginning of development and would be the best intigration.

**Comment by potrudeau** [comment]
Hi @monza258 !
Doesn't pushcut have native Homekit integration ?

**Comment by monza258** [comment]
Hi,

There is no Plugin yet. Now I use the http-Webhook Plugin to integrate pushbuttons to send notifications with pushcut in homebridge.

**Comment by potrudeau** [comment]
I did a bit of reading! Pushcut is alot like Pushover, but gives you more native integrations and has nice ties with the Apple ecosystem. Pushover is like DIY! :)

homebridge-http-webhooks is a very nice plugin. But it's overly complex for your needs I guess. 

I could add pushcut's webhook to this plugin.
![image](https://user-images.githubusercontent.com/1062160/80136278-7dffc680-856f-11ea-9063-de0c27eb7b70.png)

Or, we could use the Web API and send custom notification. See [API](https://www.pushcut.io/webapi.html).

**Comment by monza258** [comment]
Yeah, Pushcut is my favorite app. Much more options to Automat my home. If you include it, you’re the first that include Pushcut to Homebridge. 😉 

Pushcut is constantly being developed.

**Comment by potrudeau** [comment]
Right now, you're using Webhook ?

**Comment by monza258** [comment]
> Right now, you're using Webhook ?

Yes

**Comment by potrudeau** [comment]
> > Right now, you're using Webhook ?
> 
> Ye.

I could implement this or WebAPI. I will start and create an account. 
I'll get this done after I published the IFTTT messages,

**Comment by monza258** [comment]
Hi, Have you already started?

**Comment by potrudeau** [comment]
> Hi, Have you already started?

Yes ;)

**Comment by monza258** [comment]
> > Hi, Have you already started?
> 
> 
> 
> Yes ;)

Hi, any new details? I'm waiting excitedly. 😅

**Comment by raixer** [comment]
Even tho you can invoke the pushcut webhook directly from homekit using a home shortcut automation having more options is good. And I like the idea so far to have a single plugin for all messaging needs. 

This plugin is excellent. And the config UI makes it really easy to manage.

**Comment by potrudeau** [comment]
@monza258 : I haven't forgotten. I'm currently very busy with work. Also here, in Canada, we only get nice weather a few months/weeks per year, so i'm currently outside more then in front of my computer! ;) It's coming!

@raixer : That's the idea. I was tired of having 3-5 plugins for all my needs. The next update will also feature improvements to the config UI.

**Comment by monza258** [comment]
> @monza258 : I haven't forgotten. I'm currently very busy with work. Also here, in Canada, we only get nice weather a few months/weeks per year, so i'm currently outside more then in front of my computer! ;) It's coming!
> 
> 
> 
Hi, is there any chance to include Pushcut?

**Comment by potrudeau** [comment]
The code is done. Missing only UI.
I will release a version this week where you could test it, if you configure it using the config file. 
Sounds good ?

**Comment by potrudeau** [comment]
@monza258 : 
Good news: Pushcut works in version 0.0.8. I will add more of the features to this beta version, before doing the UI. In the meantime, you can see how to use it [here](https://github.com/potrudeau/homebridge-messenger/blob/master/configuration-examples/pushcut.example.json). Feel free to test it! :)

**Comment by monza258** [comment]
Wow, thank you. Can you also include the option to send a image and select the device? 

There are more options in Pushcut Api.
image 
devices
sound

**Comment by potrudeau** [comment]
> Wow, thank you. Can you also include the option to send a image and select the device?
> 
> There are more options in Pushcut Api.
> image
> devices
> sound

I know. That's what I meant by adding features. These will be added, as well as the possibility to sent JSON such as actions.

**Comment by monza258** [comment]
Ok, I will install the beta and I test it. Thanks 🙏

**Comment by potrudeau** [comment]
> Ok, I will install the beta and I test it. Thanks 🙏

Let me know if something doesn't work.

**Comment by monza258** [comment]
Hi @potrudeau, it works fine. My important Future is images and select device. Wait for more. Thank you to develop it.

**Comment by monza258** [comment]
Hi @potrudeau , thank for implementing Pushcut. It works fine. Do you have check about implementing Image option in Pushcut? Then I can send Images from my IP Cam with your message plugin an Pushcut. Rgds Denis

**Comment by monza258** [comment]
Hi @potrudeau, can you pls. include image Option?

**Comment by ghost** [comment]
Any update on this? Pushcut now supported?
